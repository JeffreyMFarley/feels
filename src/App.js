import './App.less'
import Mood from './components/Mood'
import React from 'react'

/**
* renders the App component
*
* @returns {string} the HTML to render
*/
function App() {
  return (
    <div id="App">
      <header>Header</header>
      <main>
        <nav>
          <Mood />
        </nav>
        <section>
          <article>Article 1</article>
          <article>Article 2</article>
          <article>Article 3</article>
        </section>
        <aside>ads</aside>
      </main>
      <footer>footer</footer>
    </div>
  )
}

export default App;
