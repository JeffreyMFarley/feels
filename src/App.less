body {
    main {
        display: flex;
        flex-direction: row;
    }

    section {
        border: solid 1px red;

        flex: 0 0 60vw;
        flex-direction: column;
        flex-wrap: wrap;

        article {
            flex: 1;
        }
    }

    aside {
        border: solid 1px green;

        flex: 0 0 19vw;
    }

    nav {
        border: solid 1px blue;

        flex: 0 0 19vw;

        ul {
            margin: 0;
            padding: 0;

            display: flex;
            flex-direction: row;
            flex-wrap: wrap;

            li {
                display: flex;
                flex: 1;
            }
        }
    }

    header, footer {
    }
}