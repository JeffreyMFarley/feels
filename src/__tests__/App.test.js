import App from '../App';
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import React from 'react';
import renderer from 'react-test-renderer';

it( 'renders correctly', () => {
  const mockStore = configureMockStore( [] )
  const store = mockStore( {
    mood: { foo: 1, bar: 2 }
  } )

  const tree = renderer
    .create( <Provider store={store}><App /></Provider> )
    .toJSON();
  expect( tree ).toMatchSnapshot();
} );
