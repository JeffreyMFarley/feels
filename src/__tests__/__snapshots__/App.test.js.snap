// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="App"
>
  <header>
    Header
  </header>
  <main>
    <nav>
      <div>
        <label
          hmtlfor="mood_foo"
        >
          foo
        </label>
        <input
          id="mood_foo"
          max="5"
          min="0"
          name="mood_foo"
          onChange={[Function]}
          type="range"
          value={1}
        />
      </div>
      <div>
        <label
          hmtlfor="mood_bar"
        >
          bar
        </label>
        <input
          id="mood_bar"
          max="5"
          min="0"
          name="mood_bar"
          onChange={[Function]}
          type="range"
          value={2}
        />
      </div>
    </nav>
    <section>
      <article>
        Article 1
      </article>
      <article>
        Article 2
      </article>
      <article>
        Article 3
      </article>
    </section>
    <aside>
      ads
    </aside>
  </main>
  <footer>
    footer
  </footer>
</div>
`;
